ops/cli.py

```diff
@@ -1,351 +1,343 @@
+
+        if not input(f"Are you sure you want to deploy version {self.version}? [N/y]").lower().startswith("y"):
+            exit(0)
+
+        self.__set_up_helm()
+
+        self.__publish_docker_images()
 
         deploy = os.path.join("derived", "helm", "deploy")
         if not os.path.exists(deploy):
             os.makedirs(deploy)
 
         values = "wfl-values.yaml"
-        ctmpl = f"derived/2p/gotc-deploy/deploy/{gotc_folder}/helm/{values}.ctmpl"
+        ctmpl = f"derived/2p/{self.gotc_deploy_folder}/helm/{values}.ctmpl"
         shutil.copy(ctmpl, deploy)
 
-        render_ctmpl(ctmpl_file=f"{deploy}/{values}.ctmpl", WFL_VERSION=version)
-        helm_deploy_wfl(values=f"{deploy}/{values}")
+        render_ctmpl(ctmpl_file=f"{deploy}/{values}.ctmpl", WFL_VERSION=self.version)
+        self.__helm_deploy_wfl(values=f"{deploy}/{values}")
 
-        container = run_cloudsql_proxy(project=project, cloudsql_instance_name="zero-postgresql")
-        with open(f"{deploy}/{values}") as f:
-            helm_values = yaml.safe_load(f)
+        container = self.run_cloudsql_proxy()
+        with open(f"{deploy}/{values}") as values_file:
+            helm_values = yaml.safe_load(values_file)
             env = helm_values['api']['env']
-            run_liquibase_migration(env['ZERO_POSTGRES_USERNAME'], env['ZERO_POSTGRES_PASSWORD'])
-            info("=> Stopping cloud_sql_proxy")
-            shell(f"docker stop {container}")
+            self.__run_liquibase_migration(env['ZERO_POSTGRES_USERNAME'], env['ZERO_POSTGRES_PASSWORD'])
+        info("=>  Stopping cloud_sql_proxy")
+        shell(f"docker stop {container}")
+
+        success("Deployment is done!")
+        info(shell("kubectl get pods"), plain=True)
 
-        success("[✔] Deployment is done!")
-        shell("kubectl get pods")
         return 0
 
+    @staticmethod
+    def __set_up_helm():
+        """Add helm repo and update."""
+        info("=>  Setting up Helm charts")
+        shell("helm repo add gotc-charts https://broadinstitute.github.io/gotc-helm-repo/")
+        shell("helm repo update")
+        success("Set up Helm charts")
+
+    def __publish_docker_images(self):
+        """Publish docker build."""
+        info(f"=>  Publishing Docker images for version {self.version}")
+        for module in ["api", "ui"]:
+            shell(f"docker push broadinstitute/workflow-launcher-{module}:{self.version}")
+        success("Published Docker images")
+
+    def __helm_deploy_wfl(self, values: str):
+        """Helm deploy using gotc-charts/wfl."""
+        info(f"=>  Deploying to {self.cluster_name} in {self.cluster_namespace} namespace")
+        info("    This must run on a non-split VPN", plain=True)
+        shell(f"helm upgrade wfl-k8s gotc-charts/wfl -f {values} --install")
+        success("WFL deployed")
+
+    @staticmethod
+    def __run_liquibase_migration(db_username: str, db_password: str):
```

Comment:
Again there's no reason for these to be static methods. I have no idea why they exist in python.

Reply:
Resolved in new version, no methods exist in the file.